tensorflow/core/kernels/reduction_gpu_kernels.cu.h

```diff
@@ -64,59 +82,103 @@ struct Square {
   }
 };
 
+#if TENSORFLOW_USE_ROCM
+template <>
```

Comment:
I don't think we should need all separate code for ROCM.

Reply:
See the comment above